src/app/shared/table-of-contents/table-of-contents.ts

```diff
@@ -0,0 +1,121 @@
+import {
+  Component, Input, AfterViewInit, Inject, ElementRef, OnInit
+} from '@angular/core';
+import {DOCUMENT} from '@angular/platform-browser';
+import {Router, ActivatedRoute, Event, NavigationEnd} from '@angular/router';
+import {Observable} from 'rxjs/Observable';
+import {Subscription} from 'rxjs/Subscription';
+import 'rxjs/add/observable/fromEvent';
+import 'rxjs/add/operator/debounceTime';
+
+@Component({
+  selector: 'table-of-contents',
+  styleUrls: ['./table-of-contents.scss'],
+  template: `
+    <div *ngIf="links?.length" class="toc-container">
+      <h2>Contents</h2>
+      <ul>
+        <li *ngFor="let link of links; let i = index" class="li-{{link.type}}">
+          <a [href]="_rootUrl + '#' + link.id"
+            [class.active]="link.active">
+            {{link.name}}
+          </a>
+        </li>
+      </ul>
+    </div>
+  `
+})
+export class TableOfContents implements OnInit, AfterViewInit {
+
+  @Input() links: any[] = [];
```

Comment:
Can this be something other than `any`? Using it in this manner is pretty strongly discouraged 
(here and elsewhere)